### DISCUSSION TITLE
Unable to render swagger [Feb 11]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5157] [Discord Threads]
> 🙋‍♂️ **Joecode**  
## Whenever I try and view swagger after connecting to an existing db/schema I get this error. I am using version 0.101.0-beta.0. If the project only contains tables created by nocodb, the swagger works. To reproduce: 1. Add a new data source to the project and connect to an existing db/schema 2. Try and view the swagger page. Any ideas what might cause this?  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1062842372246999143/image.png)

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### are you accessing locally? or you are using docker?  
#  
> 👨‍💻 **pranavxc**  
#### you have to login in-order to access the swagger  
#  
> 🙋‍♂️ **Joecode**  
#### I am logged in, running the dev setup, so frontend on port 3000, backend on 8080. The link to swagger takes me to localhost:8080/api/v1/db/meta/projects/p_<project_id>/swagger. That is when I see the Unable to render this definition message  
#  
> 🙋‍♂️ **Joecode**  
#### Does the swagger version need to be setup in the connection json?  
#  
> 🙋‍♂️ **wkw**  
#### what pranav meant is   
####    
#### - go to http://localhost:8080/dashboard and login   
#### - go back to http://localhost:3000/dashboard to access swagger   
####    
#### this would solve your problem  
#  
> 🙋‍♂️ **wkw**  
#### the error message is misleading. the root cause is you need to authorize in 8080 side first.  
#  
> 🙋‍♂️ **Joecode**  
#### Ah- makes sense. Thank you!  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1062842372607721643)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
